src/libcore/tests/test_xml.py

```diff
@@ -297,3 +298,486 @@ def test20_upgrade_tree(variant_scalar_rgb):
                                <float name="intIOR" value="1.33"/>
                            </bsdf>
                        </scene>""")
+
+
+def test21_dict_plugins(variants_all_rgb):
```

Comment:
Let's put these into a separate file (e.g. `test_dict.py`) since it's not really related to XML, and the original test file is already quite large.